File: 23-03-2026/coffee_roast_detector.py
import re

def detect_roast(beans):
    apos = len(re.findall(r"\'",beans))
    dash = len(re.findall(r"\-",beans)) * 2 
    dot = len(re.findall(r"\.",beans))  * 3
    total = sum([dash, apos, dot]) / len(beans)

    if total < 1.75:
        return "Light"
    elif total <= 2.5 and total >= 1.75:
        return "Medium"
    else:
        return "Dark"

print(detect_roast("''-''''''-'-''--''''")) # "Light".
print(detect_roast(".'-''-''..'''.-.-''-")) # "Medium".
print(detect_roast("--.''--'-''.--..-.--")) # "Medium".
print(detect_roast("-...'-......-..-...-")) # "Dark".
print(detect_roast(".--.-..-......----.'")) # "Medium".
print(detect_roast("..-..-..-..-....-.-.")) # "Dark".
print(detect_roast("-'-''''''..-'.''-'.'")) # "Light".

File: 23-03-2026/coffee_roast_detector.js
function detectRoast(beans) {
  const apos = beans.match(/\'/g)?.length || 0;
  const dash = (beans.match(/\-/g)?.length || 0) * 2;
  const dot = (beans.match(/\./g)?.length || 0) * 3;
  const total = (apos + dash + dot) / beans.length;

  return total < 1.75
    ? "Light"
    : total <= 2.5 && total >= 1.75
      ? "Medium"
      : "Dark";
}

console.log(detectRoast("''-''''''-'-''--''''")); // "Light".
console.log(detectRoast(".'-''-''..'''.-.-''-")); // "Medium".
console.log(detectRoast("--.''--'-''.--..-.--")); // "Medium".
console.log(detectRoast("-...'-......-..-...-")); // "Dark".
console.log(detectRoast(".--.-..-......----.'")); // "Medium".
console.log(detectRoast("..-..-..-..-....-.-.")); // "Dark".
console.log(detectRoast("-'-''''''..-'.''-'.'")); // "Light".
